atlasdb-autobatch/src/main/java/com/palantir/atlasdb/autobatch/DisruptorAutobatcher.java

```diff
@@ -187,6 +212,10 @@ static <T, R> DisruptorAutobatcher<T, R> create(
         disruptor.handleEventsWith(
                 (event, sequence, endOfBatch) -> eventHandler.onEvent(event.consume(), sequence, endOfBatch));
         disruptor.start();
-        return new DisruptorAutobatcher<>(disruptor, disruptor.getRingBuffer(), safeLoggablePurpose, closingCallback);
+        return new DisruptorAutobatcher<>(
+                disruptor,
+                disruptor.getRingBuffer(),
+                AutobatcherTelemetryComponents.create(safeLoggablePurpose, new DefaultTaggedMetricRegistry()),
```

Comment:
I am a little bit paranoid about using `new DefaultTaggedMetricRegistry()`. Will metrics be published in prod? Will they have a service tag?

We cannot always hook a metrics manager here (used for SnapshotTransaction and possibly gets more info from the relevant service). In this same package, `BatchSizeRecorder` uses `SharedTaggedMetricRegistries.getSingleton()` (this is deprecated).